main.py

```diff
@@ -1,139 +1,283 @@
-        if read_current_mode() != current_mode:
-            return
-        pixels[r]=colorwheel((255/20*(r+movingRainbow))%255)  
+        wait_and_check(waittime)
+
+    # Bright -> Dark
+    for step in range(steps, 0, -1):
+        for pixel in range(NUM_PIXELS):
+            pixels[pixel] = (r*step/steps, g*step/steps, b*step/steps)
         pixels.show()
-    movingRainbow = movingRainbow+1
+        wait_and_check(waittime)
 
-'''
-#############################################
-######  get current input from robot
-#############################################
-'''
 
-def read_current_mode():
-    # TODO: read from serial or input pins
-    # for now just statically returning disabled
-    return 31
+def static():
+    for pixel in range(NUM_PIXELS):
+        pixels[pixel] = (r, g, b)
+    pixels.show()
+    time.sleep(waittime)
 
 
-class ModeChangedException(Exception):
-    pass
+# Runs when robot is enabled
+def enabled():
+    match light_state:
+        case LightDisplayState.Static:
+            static()
+
+        case LightDisplayState.Flashing:
+            flash()
+        
+        case _:
+            print("Light state not matched")
+ 
 
-def wait_and_check(durationS):
-    if mode == read_current_mode():
-        time.sleep(durationS)
+def wait_and_check(waittime: float):
+    # Robot updates every "waittime", so our frames per second is 1/waittime, waittime = 0.04 means 25 fps
+    if robot_state == get_robot_state() and robot_mode == get_robot_mode():
+        time.sleep(waittime)
     else:
-        raise ModeChangedException("Mode changed")
+        raise StateChangedException("Robot state or mode changed")
+            
 
-#############################################
-######  main loop
-#############################################
+def get_robot_state():
+    return RobotStateEnum.NotHoldingNote
 
+
+def get_robot_mode():
+    return RobotMode.Enabled
+
+
+# Main function
 def main():
-    global mode
-    # Read mode and alliance from pins
-    mode = read_current_mode()
-
-    try:    
-        #Select display option
-        if mode == 31:
-            no_code()
-        elif mode == 1:
-            disabled()
-        elif mode == 2:
-            enabled(mode)
-        elif mode == 3:
-            moving_rainbow(False)
-        elif mode == 4:
-            disabled_with_auto()
-        elif mode == 5: #arm in position
-            moving_rainbow(True)
-        else:
-            no_code()
-    except ModeChangedException:
-        # Mode changed, restart loop
+    # Assuming we are getting a state
+    # Also, "action" or "rainbow" states should have higher priority than static
+    global robot_state
+    global robot_mode
+    robot_state = get_robot_state()
+    robot_mode = get_robot_mode()
+
+    # Run the lights (actual displaying)
+    try:
+        match robot_mode:
+            case RobotMode.NoCode:
+                # Runs when there is no code on the RoboRIO, STATIC BLACK
```

Comment:
⭐  Using 'black' which is basically no lights on means it's impossible to tell the difference between this state and the lights just being totally broken and not working?